Library/Homebrew/software_spec.rb

```diff
@@ -53,7 +53,7 @@ def owner=(owner)
     @resource.owner = self
     resources.each_value do |r|
       r.owner     = self
-      r.version ||= version
+      r.version ||= version.dup
```

Comment:
I think the `dup` is cleaner and safer, not to mention that we don't want to instantiate `HeadVersion` directly (only via `Version.create`), but that's just a detail.